stencil-library/src/components/business-form/business-form-stepped.tsx

```diff
@@ -0,0 +1,183 @@
+import { Component, Host, h, Prop, State, Watch } from '@stencil/core';
+import { FormController } from '../form/form';
+import businessFormSchema from './business-form-schema';
+import { Api } from '../../api';
+import { parseForPatching } from './helpers';
+
+
+const componentStepMapping = {
+  0: (formController) => <justifi-business-generic-info formController={formController} />,
+  1: (formController) => <justifi-legal-address-form formController={formController} />,
+  2: (formController) => <justifi-additional-questions formController={formController} />,
+  3: (formController) => <justifi-business-representative formController={formController} />,
+  4: (formController) => <justifi-business-owners formController={formController} />
+};
+
+
+/**
+ * @exportedPart label: Label for inputs
+ * @exportedPart input: The input fields
+ * @exportedPart input-invalid: Invalid state for inputfs
+ */
+@Component({
+  tag: 'justifi-business-form-stepped',
+  styleUrl: 'business-form-stepped.scss',
+})
+export class BusinessFormStepped {
+  @Prop() authToken: string;
+  @Prop() businessId?: string;
+  @State() isLoading: boolean = false;
+  @State() currentStep: number = 0;
+  @State() totalSteps: number = 4;
+  @State() percentageComplete: number = 0;
```

Comment:
I suggest refactoring percentageComplete from a state to a computed getter for simplicity and maintainability:

```
get percentageComplete() {
  return (this.currentStep / this.totalSteps) * 100;
}

```

This change will let us:

Remove the @Watch on currentStep.
Drop the updatePercentageCompleteOnStepChange method.
Also drop the percentageComplete state.
Let's keep derived values like this computed to ensure accuracy.

Thoughts?

Reply:
Sure, good suggestion!